astroid/nodes/scoped_nodes/scoped_nodes.py

```diff
@@ -1469,6 +1485,44 @@ def is_generator(self) -> bool:
         """
         return bool(next(self._get_yield_nodes_skip_lambdas(), False))
 
+    def _infer(
+        self, context: InferenceContext | None = None, **kwargs: Any
+    ) -> Generator[objects.Property | FunctionDef, None, InferenceErrorInfo]:
+        from astroid import objects  # pylint: disable=import-outside-toplevel
```

Comment:
I think this is a case of pretty bad cyclic imports. With this approach we can't avoid this I think which makes me doubtful about whether this is the right approach.

Reply:
Challenge accepted!

[Here is a branch](https://github.com/pylint-dev/astroid/compare/main...jacobtylerwalls:astroid:move-property-and-partial?expand=1) that removes these cyclic imports by moving `PartialFunction` and `Property` to `scoped_nodes`, which makes sense, since they inherit from `FunctionDef` anyway (what's the use of a Property being an object, anyway?)

The ease with which I could do this gives me hope that this is a _good_ direction that can enable easier refactors in iterative PRs. What do you think?

Reply:
If we go with the `infer_object` approach, we're freezing the inference/objects/protocols ontologies in place, instead of actually questioning if everything is where it should be, which we should feel free to do during breaking changes season ☔

Reply:
See my comment on the other pull request. I don't necessarily think we would be freezing anything in place.

I do think that we should merge your `PartialFunction` branch to `master` irrespective of the approach. Seems like a very sensible change!

Reply:
Maybe bursting the big node files is actually (one of) the way to solve circular imports?